specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json

```diff
@@ -0,0 +1,1024 @@
+        },
+        "policySetDefinitionId": {
+          "description": "Policy set definition ID, if the policy assignment is for a policy set.",
+          "type": "string"
+        },
+        "policySetDefinitionName": {
+          "description": "Policy set definition name, if the policy assignment is for a policy set.",
+          "type": "string"
+        },
+        "policySetDefinitionOwner": {
+          "description": "Policy set definition owner, if the policy assignment is for a policy set.",
+          "type": "string"
+        },
+        "policySetDefinitionCategory": {
+          "description": "Policy set definition category, if the policy assignment is for a policy set.",
+          "type": "string"
+        },
+        "policySetDefinitionParameters": {
+          "description": "Policy set definition parameters, if the policy assignment is for a policy set.",
+          "type": "string"
+        },
+        "managementGroupIds": {
+          "description": "Comma separated list of management group IDs, which represent the hierarchy of the management groups the resource is under.",
+          "type": "string"
+        },
+        "policyDefinitionReferenceId": {
+          "description": "Reference ID for the policy definition inside the policy set, if the policy assignment is for a policy set.",
+          "type": "string"
+        },
+        "complianceState": {
+          "description": "Compliance state of the resource.",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "Tenant ID for the policy event record.",
+          "type": "string"
+        },
+        "principalOid": {
+          "description": "Principal object ID for the user who initiated the resource operation that triggered the policy event.",
+          "type": "string"
+        },
+        "components": {
+          "description": "Components event records. By default this is not populated.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ComponentEventDetails"
+          }
+        }
+      },
+      "additionalProperties": {
+        "description": "The list of key/value pairs for the additional properties, in the format 'key':'value' where key = the field name, and value = the field value. By default this is not populated"
+      }
+    },
+    "ComponentEventDetails": {
+      "type": "object",
+      "description": "Component event details.",
+      "properties": {
+        "id": {
+          "description": "Component Id.",
+          "type": "string"
+        },
+        "type": {
+          "description": "Component type.",
+          "type": "string"
+        },
+        "name": {
+          "description": "Component name.",
+          "type": "string"
+        },
+        "timestamp": {
+          "description": "Timestamp for component policy event record.",
+          "type": "string",
+		  "format": "date-time"
+        },
+		"tenantId": {
```

Comment:
>"tenantId" [](start = 2, length = 10)

This can be different than the root policyEvent tenantId?

Reply:
Yes, the user for keyvault can be different from the user operating over certificates that belong to given keyvault.